packages/alfa-rules/src/sia-r68/rule.ts

```diff
@@ -97,3 +103,37 @@ function isRequiredChild(
     };
   }
 }
+
+/**
+ * Collect all descendants of node that:
+ * * match includeNode; and
+ * * do not have an ancestor matching excludeSubtree; and
+ * * have an ancestor matching collectSubtree
+ */
+function visit<T extends Node = Node>(
```

Comment:
This is meant to be generic enough to re-factor `visit` in R83 (and maybe other). But I want feedback before refactoring 😉